### PR TITLE
Add Consul Service mesh adaptor options to helm chart

### PR DESCRIPTION
**Pull request by nicholasjackson**
**Description**

This PR adds capability to install the Consul Service mesh adaptor automatically using the Helm chart. 

I followed the patterns from Istio and LinkerD to implement this change, have tested manually.

Kind regards,

Nic

### CODE REVIEW

**Review by leecalcote — DISMISSED** [review]
Thank you, @nicholasjackson. 👍 Looks real good to me.

File: install/kubernetes/helm/meshery/charts/mesheryconsul/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+apiVersion: v2
+name: mesheryconsul
+description: Meshery Adapter for Consul chart.
+source: 
+  - https://github.com/layer5io/meshery-istio
```
Comment (mgfeller) [inline install/kubernetes/helm/meshery/charts/mesheryconsul/Chart.yaml:5]: https://github.com/layer5io/meshery-consul

File: install/kubernetes/helm/meshery/charts/mesheryconsul/values.yaml [hunk 2]
```diff
@@ -0,0 +1,90 @@
+# Default values for mesheryistio.
```
Comment (mgfeller) [inline install/kubernetes/helm/meshery/charts/mesheryconsul/values.yaml:1]: Default values for mesheryconsul

File: install/kubernetes/helm/meshery/charts/mesheryconsul/templates/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
+      serviceAccountName: {{ include "mesheryconsul.serviceAccountName" . }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      containers:
+        - name: {{ .Chart.Name }}
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          # TODO: Need to confirm the mesheryistio image version
```
Comment (mgfeller) [inline install/kubernetes/helm/meshery/charts/mesheryconsul/templates/deployment.yaml:31]: mesheryconsul

File: install/kubernetes/helm/meshery/charts/mesheryconsul/templates/NOTES.txt [hunk 4]
```diff
@@ -0,0 +1,21 @@
+1. Get the application URL by running these commands:
+{{- if .Values.ingress.enabled }}
+{{- range $host := .Values.ingress.hosts }}
+  {{- range .paths }}
+  http{{ if $.Values.ingress.tls }}s{{ end }}://{{ $host.host }}{{ . }}
+  {{- end }}
+{{- end }}
+{{- else if contains "NodePort" .Values.service.type }}
+  export NODE_PORT=$(kubectl get --namespace {{ .Release.Namespace }} -o jsonpath="{.spec.ports[0].nodePort}" services {{ include "mesheryistio.fullname" . }})
```
Comment (mgfeller) [inline install/kubernetes/helm/meshery/charts/mesheryconsul/templates/NOTES.txt:9]: mesheryconsul.fullname 
(several occurrences)

File: install/kubernetes/helm/meshery/charts/mesheryconsul/values.yaml [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+replicaCount: 1
+
+image:
+  repository: layer5/meshery-consul:latest
```
Comment (mgfeller) [inline install/kubernetes/helm/meshery/charts/mesheryconsul/values.yaml:8]: @leecalcote should this be `stable-latest`? It is in .meshery/meshery.yaml

**Review by mgfeller — CHANGES_REQUESTED** [review]
This looks great, @nicholasjackson! :+1: 
"istio" should be replaced by "consul" in a couple of places (commented) and I'm not quite sure about the image tag to be used, included @leecalcote for clarification.

**Review by Aisuko — APPROVED** [review]
approve

**Comment by leecalcote** [comment]
A question (directed to those participating in the mutiny I see in this PR): why don't my PRs get ❤️ s and 🎉 s like when @nicholasjackson submits a PR?

File: install/kubernetes/helm/meshery/charts/mesheryconsul/templates/deployment.yaml (hunk 3 above)
Comment (leecalcote) [reply install/kubernetes/helm/meshery/charts/mesheryconsul/templates/deployment.yaml:31]: Good catch. Yes, this tag needs to be `stable-latest`. I suspect this update is needed across all other charts, too.

File: install/kubernetes/helm/meshery/charts/mesheryconsul/values.yaml (hunk 5 above)
Comment (leecalcote) [reply install/kubernetes/helm/meshery/charts/mesheryconsul/values.yaml:8]: Good catch. Yes, this tag needs to be stable-latest. I suspect this update is needed across all other charts, too.

File: install/kubernetes/helm/meshery/charts/mesheryconsul/templates/NOTES.txt (hunk 4 above)
Comment (leecalcote) [reply install/kubernetes/helm/meshery/charts/mesheryconsul/templates/NOTES.txt]: Updated each

File: install/kubernetes/helm/meshery/charts/mesheryconsul/templates/deployment.yaml (hunk 3 above)
Comment (leecalcote) [reply install/kubernetes/helm/meshery/charts/mesheryconsul/templates/deployment.yaml]: Updated

File: install/kubernetes/helm/meshery/charts/mesheryconsul/Chart.yaml (hunk 1 above)
Comment (leecalcote) [reply install/kubernetes/helm/meshery/charts/mesheryconsul/Chart.yaml]: An important item. Updated.

**Review by leecalcote — APPROVED** [review]
Cleaning up after @nicholasjackson... my work is never over.

**Comment by leecalcote** [comment]
Oh, no. We need to fix the Welcome Bot. @nicholasjackson you should have gotten one of these:

Screen Shot 2020-09-14 at 3 08 45 PM

Yay! @nicholasjackson 🎈

**Pull request merged**